### DISCUSSION TITLE
about CancellationToken in AspNetUserManager.cs

### DISCUSSION THREAD

**Discussion by HosseinBabmoradi** [dotnet/aspnetcore#48639] [Q&A] [answered]
Hi 
The cancellationToken in UserManager class never change 
i wrote these methods in AspNetUserManager 
`public CancellationToken GetBaseCancellationToken() => base.CancellationToken;`
`public CancellationToken GetCancellationToken() => _cancel;`

codes in razor file 
```
public async Task OnGetAsync(CancellationToken cancellationToken)
{
        cancellationToken = new CancellationToken(true);
        await aspNetUserManager.GetUserAsync(HttpContext.User, cancellationToken);
        var baseCancellationToken = aspNetUserManager.GetBaseCancellationToken();
        //return false
        var CancellationToken = aspNetUserManager.GetCancellationToken();
        //return true
}
```

UserManager methods work with base.CancellationToken and it always is false

**Comment by martincostello** [comment]
Have you registered `IHttpContextAccessor` in the DI container for your app using [`services.AddHttpContextAccessor()`](https://github.com/dotnet/aspnetcore/blob/e4c6621d9a09cae8565ad918ba5083c9b28840bd/src/Http/Http/src/HttpServiceCollectionExtensions.cs#L19)?

It's needed to get the current HTTP request's `CancellationToken`:

https://github.com/dotnet/aspnetcore/blob/e4c6621d9a09cae8565ad918ba5083c9b28840bd/src/Identity/Core/src/AspNetUserManager.cs#L42

**Reply by HosseinBabmoradi** [reply]
it`s still returning false

**Reply by martincostello** [reply]
What property are you actually checking that's returning the booleans? `CanBeCancelled` or `IsCancellationRequested`?

**Reply by HosseinBabmoradi** [reply]
`IsCancellationRequested`

**Reply by martincostello** [reply]
What are you doing that means that the current HTTP request should be being marked as cancelled?

**Reply by HosseinBabmoradi** [reply]
I want to change default value of CancellationToken in UserManager  class
but it`s readonly and in AspNetUserManager it override the CancellationToken but the value of base.CancellationToken never changed

**Comment by HosseinBabmoradi** [comment, marked as answer]
I got my answer here [stackoverflow](https://stackoverflow.com/questions/76411854/why-dont-async-methods-in-identity-services-in-asp-net-core-have-a-parameter-fo)
